ketcher-autotests/tests/Structure-Creating-&-Editing/Bond-Tool/all-bonds.spec.ts

```diff
@@ -274,5 +308,254 @@ for (const bondToolKey of Object.keys(BondTool)) {
+       */
+      await selectNestedTool(page, BondTool[bondToolKey]);
+      await clickInTheMiddleOfTheScreen(page);
+    });
+
+    test.describe('Saving and rendering', () => {
+      /**
+       *   Test cases: EPMLSOPKET-1378, 1386, 1395, 1401, 1409, 1415, 1421, 1427, 1433, 1442, 1449, 1456, 2243, 2249
+       */
+      const fileName = `saving-and-rendering-${bondToolKey}-bond.mol`;
+      test(`Save to file`, async () => {
+        await selectNestedTool(page, BondTool[bondToolKey]);
+        await clickOnTheCanvas(page, -200, 0);
+        await clickInTheMiddleOfTheScreen(page);
+        await selectTopPanelButton(TopPanelButton.Save, page);
+        await page.getByRole('button', { name: 'Save', exact: true }).click();
+      });
+
+      test(`Open and edit`, async () => {
+        await openFileAndAddToCanvas(fileName, page);
+        await selectLeftPanelButton(LeftPanelButton.ReactionPlusTool, page);
+        await clickOnTheCanvas(page, 200, 0);
+      });
+    });
+
+    test(`Check that ${bondToolKey} bond between atoms are centered and drawn symmetrically`, async () => {
+      /**
+       *Test case: EPMLSOPKET-16931
+       *Description: Check that Bonds between atoms are centered and drawn symmetrically
+       */
+      await selectNestedTool(page, BondTool[bondToolKey]);
+      await clickInTheMiddleOfTheScreen(page);
+    });
+  });
+}
+
+test.describe('Bond Tool', () => {
+  const toolsForTest = [BondTool.SINGLE, BondTool.DOUBLE, BondTool.TRIPPLE];
+  test.beforeEach(async ({ page }) => {
+    await page.goto('', { waitUntil: 'domcontentloaded' });
+  });
+
+  test.afterEach(async ({ page }) => {
+    await takeEditorScreenshot(page);
+  });
+
+  for (const tool of toolsForTest) {
+    test(`Functional Group with attach ${tool}`, async ({ page }) => {
+      /**
+       *Test case: EPMLSOPKET-10086
+       *Description: A bond is added to a contracted functional group and form a bond
+       */
+      await pressButton(page, STRUCTURE_LIBRARY_BUTTON_NAME);
+      await page.getByRole('tab', { name: 'Functional Groups' }).click();
+      await selectFunctionalGroups(FunctionalGroups.Boc, page);
+      await clickInTheMiddleOfTheScreen(page);
+      await selectNestedTool(page, tool);
+      await clickInTheMiddleOfTheScreen(page);
+    });
+
+    test(`Creating two (or more) connected ${tool} bonds`, async ({ page }) => {
+      /**
+       * Test cases: EPMLSOPKET - 2920/2921
+       */
+      await clickInTheMiddleOfTheScreen(page);
+      await selectNestedTool(page, tool);
+      await clickInTheMiddleOfTheScreen(page);
+      await clickInTheMiddleOfTheScreen(page);
+    });
+  }
+
+  test('Drop down list: verification', async ({ page }) => {
+    /**
+     *Test case: EPMLSOPKET-1366
+     *Description: Drop down list: verification
+     */
+    await selectLeftPanelButton(LeftPanelButton.SingleBond, page);
+    await selectLeftPanelButton(LeftPanelButton.SingleBond, page);
```

Comment:
there's a function which allows you to open any dropdown on left panel toolbar (so you don't have to do double clicks)
![image](https://github.com/epam/ketcher/assets/139877378/21b17b61-f4ab-4d9c-bb82-3a01dfe446cb)

Reply:
I believe it is redundant work to create function twice. I guess we should wait until your changes will merge and use this function in the future